client/rpmtrans.c

```diff
@@ -534,6 +534,10 @@ TDNFTransAddInstallPkg(
 
 error:
     fprintf(stderr, "Error processing package: %s\n", pszPackageLocation);
+    if (!access(pszFilePath, F_OK))
```

Comment:
fix should be earlier than this. ideally during download or when download gets skipped as file exists. note that "error processing package" was not in error reports.

Reply:
From the trace, it appears that rpmReadPackageFile() call failed from TDNFTransAddInstallPkg(). Any failures from TDNFTransAddInstallPkg() API has to print "error processing package" because that's the entry point to error:. I'm wondering how this string is not printed in the error message. 
If this fix has to go earlier during the download, do you suggest to check for file stats to check the size of the file and re-download the file if file size is zero instead of failing?

Reply:
file size zero check is one of the cases. however, a complete check would be to open and check the digest. then open will fail or digest will fail if file is partially downloaded.